bundles/org.openhab.binding.yioremote/src/main/java/org/openhab/binding/yioremote/internal/YIOremoteDockHandler.java

```diff
@@ -0,0 +1,388 @@
+            }
+        };
+        websocketpollingjob = scheduler.scheduleWithFixedDelay(websocketpollingthread, 0, 30, TimeUnit.SECONDS);
+    }
+
+    protected boolean getbooleanauthenticationok() {
+        return yioremotedockhandler.booleanauthenticationok;
+    }
+
+    protected boolean getbooleanheartbeat() {
+        boolean booleanresult = yioremotedockhandler.booleanheartbeat;
+        yioremotedockhandler.booleanheartbeat = false;
+        return booleanresult;
+    }
+
+    public YIOREMOTEDOCKHANDLESTATUS getyioremotedockactualstatus() {
+        return yioremotedockhandler.yioremotedockactualstatus;
+    }
+
+    public void sendMessage(YIOREMOTEMESSAGETYPE messagetype, String messagepyload) {
+        if (messagetype.equals(YIOREMOTEMESSAGETYPE.AUTHENTICATE)) {
+            yioremoteDockwebSocketClient.sendMessage("{\"type\":\"auth\", \"token\":\"" + messagepyload + "\"}");
+            logger.debug("sending authenticating message: \"{\"type\":\"auth\", \"token\":\"{}\"}\"", messagepyload);
+        } else if (messagetype.equals(YIOREMOTEMESSAGETYPE.HEARTBEAT)) {
+            stringlastsendircode = "\"0;0x0;0;0\"";
+            yioremoteDockwebSocketClient.sendMessage(
+                    "{\"type\":\"dock\", \"command\":\"ir_send\",\"code\":\"0;0x0;0;0\", \"format\":\"hex\"}");
+            logger.debug(
+                    "sending heartbeat message: {\"type\":\"dock\", \"command\":\"ir_send\",\"code\":\"0;0x0;0;0\", \"format\":\"hex\"}");
+        } else if (messagetype.equals(YIOREMOTEMESSAGETYPE.IRRECEIVERON) && yioremotedockhandler
+                .getyioremotedockactualstatus().equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_COMPLETE)) {
+            yioremoteDockwebSocketClient.sendMessage("{\"type\":\"dock\", \"command\":\"ir_receive_on\"}");
+            logger.debug("sending IR receiver on message: {\"type\":\"dock\", \"command\":\"ir_receive_on\"");
+        } else if (messagetype.equals(YIOREMOTEMESSAGETYPE.IRRECEIVEROFF) && yioremotedockhandler
+                .getyioremotedockactualstatus().equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_COMPLETE)) {
+            yioremoteDockwebSocketClient.sendMessage("{\"type\":\"dock\", \"command\":\"ir_receive_off\"}");
+            logger.debug("sending IR receiver off message: {\"type\":\"dock\", \"command\":\"ir_receive_off\"");
+        } else if (messagetype.equals(YIOREMOTEMESSAGETYPE.IRSEND) && yioremotedockhandler
+                .getyioremotedockactualstatus().equals(YIOREMOTEDOCKHANDLESTATUS.AUTHENTICATION_COMPLETE)) {
+            yioremoteDockwebSocketClient.sendMessage("{\"type\":\"dock\", \"command\":\"ir_send\",\"code\":\""
+                    + messagepyload + "\", \"format\":\"hex\"}");
+            stringlastsendircode = messagepyload;
+            logger.debug(
+                    "sending IR message: {\"type\":\"dock\", \"command\":\"ir_send\",\"code\":\"{}\", \"format\":\"hex\"}",
+                    messagepyload);
+        }
+    }
+}
```

Comment:
You could use JSON DTOs to send and receive JSON data. See here for example: https://github.com/openhab/openhab-addons/pull/8050/files#diff-904c4bebdb6a68e4a2c9d9c8289bdffdR99

Reply:
I will look into this and keep it in mz mind for the next updated version is this okay for you?